### ISSUE TITLE
Request timeout not using abort signal

### ISSUE THREAD

**Issue by AlissonRS** [Flagsmith/flagsmith-nodejs-client#160]
In the [utils.ts file](https://github.com/Flagsmith/flagsmith-nodejs-client/blob/main/sdk/utils.ts#L65), the function below is used to send request via Fetch API:
```
        const requestWrapper = (): Promise<Response> => {
            return new Promise((resolve, reject) => {
                let timeoutId: NodeJS.Timeout;
                if (timeout) {
                    timeoutId = setTimeout(() => reject('error: timeout'), timeout);
                }
                return fetch(url, fetchOptions)
                    .then(res => resolve(res))
                    .catch(err => reject(err))
                    .finally(() => {
                        if (timeoutId) {
                            clearTimeout(timeoutId);
                        }
                    });
            });
        };
```

This is timing out the promise, but not really aborting the request, which keeps http connections open hanging.

It'd be nice if abort signal was used so the http connection is also closed.

**Comment by matthewelwell** [comment]
Thanks for raising this @AlissonRS, and submitting a PR. I've added a comment to the PR - let's continue the discussion there.

**Comment by rolodato** [comment]
This should now be fixed since 4.0.0 - let us know if this is now working for you. Thanks!
